application/apps/indexer/sources/src/binary/pcap/legacy.rs

```diff
@@ -99,6 +99,9 @@ impl<R: Read + Send + Sync> ByteSource for PcapLegacyByteSource<R> {
                         if actual_tp == *wanted {
                             Ok(Some(ReloadInfo::new(
                                 received_bytes,
+                                // BUG: This should represent all available bytes in `self.buffer`.
+                                // This assumes that the buffer will be empty on each parse call
+                                // which will fail silently when parser implementing changes.
```

Comment:
you are right...from the view of the ByteSource, not every reload-call might have discarded the previously loaded bytes.
In practice with the protocols on top of ethernet it always is a completely new payload. But the concept of the ByteSource requires us to keep track of the consumed bytes.
Please write a unit-test that can show this behavior along with another test that processes multiple UDP packets. The first one should already fail, the second one should be green. Then you can work on correcting this behavior.